worker/task.go

```diff
@@ -2166,27 +2176,33 @@ func preprocessFilter(tree *pb.FilterTree) (*facetsTree, error) {
 
 type countParams struct {
 	readTs  uint64
-	count   int64
+	counts  []int64
 	attr    string
 	gid     uint32
 	reverse bool   // If query is asking for ~pred
 	fn      string // function name
 }
 
 func (qs *queryState) evaluate(cp countParams, out *pb.Result) error {
-	count := cp.count
+	countl := cp.counts[0]
+	var counth int64
+	if cp.fn == between {
+		counth = cp.counts[1]
+	}
 	var illegal bool
 	switch cp.fn {
 	case "eq":
-		illegal = count <= 0
+		illegal = countl <= 0
 	case "lt":
-		illegal = count <= 1
+		illegal = countl <= 1
 	case "le":
-		illegal = count <= 0
+		illegal = countl <= 0
 	case "gt":
-		illegal = count < 0
+		illegal = countl < 0
 	case "ge":
-		illegal = count <= 0
+		illegal = countl <= 0
+	case "between":
+		illegal = countl <= 0 || counth <= 0
```

Comment:
another check could be countl > counth

Reply:
We are not returning any error in case of improper bounds. Returned response would be empty.

Reply:
I think we should just check that case here as we are already checking other illegal cases.

Reply:
Agreed. But the point is not to return error from here, to have between behaviour consistent.